src/Tasks/ResourceHandling/MSBuildResXReader.cs

```diff
@@ -0,0 +1,299 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Shared;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+using System.Xml.Linq;
+
+namespace Microsoft.Build.Tasks.ResourceHandling
+{
+    internal class MSBuildResXReader
+    {
+        public IReadOnlyList<IResource> Resources { get; }
+
+        public static IReadOnlyList<IResource> ReadResources(Stream s, string filename, bool pathsRelativeToBasePath)
+        {
+            var resources = new List<IResource>();
+            var aliases = new Dictionary<string, string>();
+
+            using (var xmlReader = new XmlTextReader(s))
+            {
+                xmlReader.WhitespaceHandling = WhitespaceHandling.None;
+
+                XDocument doc = XDocument.Load(xmlReader, LoadOptions.PreserveWhitespace);
+                foreach (XElement elem in doc.Element("root").Elements())
+                {
+                    switch (elem.Name.LocalName)
+                    {
+                        case "schema":
+                            // TODO: this
+                            break;
+                        case "assembly":
+                            ParseAssemblyAlias(aliases, elem);
+                            break;
+                        case "resheader":
+                            break;
+                        case "data":
+                            ParseData(filename, pathsRelativeToBasePath, resources, aliases, elem);
+                            break;
+                        default:
+                            throw new NotImplementedException();
+                    }
+                }
+            }
+
+            return resources;
+        }
+
+        private static void ParseAssemblyAlias(Dictionary<string,string> aliases, XElement elem)
+        {
+            string alias = elem.Attribute("alias").Value;
+            string name = elem.Attribute("name").Value;
+
+            aliases.Add(alias, name);
```

Comment:
This led me down a dark rabbit hole. I agree with you! But that's not at all what ResXResourceReader did: it does last-one-wins, and while I was looking at that I discovered that you don't actually need to specify the alias, because it'll infer it for you from the name 😐

Reply:
Eww.  Ok to get last in wins with a Dictionary use the indexer instead of add.